src/test/java/seedu/address/logic/commands/ListCommandTest.java

```diff
@@ -69,4 +76,80 @@ public void execute_listFiltered_beforeToday() {
                 expectedMessage, expectedModelWithPastTask);
         assertEquals(Arrays.asList(J_TASK), modelWithExtremeTemporalTasks.getFilteredTaskList());
     }
+
+    @Test
+    public void execute_listFiltered_beforeEndOfWeek() {
+        String expectedMessage = String.format(MESSAGE_TASKS_LISTED_OVERVIEW, 2);
+        ListCommand.ListFilter filter = ListCommand.ListFilter.DUE_END_OF_WEEK;
+        ListCommand command = new ListCommand(filter);
+
+        /** Get a string representation for a time almost at the end of the week. */
+        Calendar c = Calendar.getInstance();
+        c = new Calendar.Builder()
+                .setWeekDate(c.get(Calendar.YEAR), c.get(Calendar.WEEK_OF_YEAR), Calendar.SUNDAY)
+                .setTimeOfDay(20, 00, 00)
+                .build();
+        Date date = c.getTime();
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-YYYY HHmm");
```

Comment:
this date format is defined in DateFormatUtil, it's DateFormatUtil.STANDARD_FORMAT I think. You can consider using that for consistency.